pyuvdata/mwa_corr_fits.py

```diff
@@ -0,0 +1,235 @@
+            check_extra: Option to check optional parameters as well as required
+                ones. Default is True.
+            run_check_acceptability: Option to check acceptable range of the values of
+                parameters after reading in the file. Default is True.
+        """
+        
+        metafits_file=None
+        file_dict={}
+        
+        for file in filelist:
+            if file.lower().endswith('.metafits'):
+                #check that have a metafits file
+                #TODO: ***fix this so that it allows for multiple metafits files!***
+                #for now, force only one metafits file
+                if metafits_file is not None:
+                    raise ValueError('multiple metafits files in filelist')
+                metafits_file = file
+            #organize data files
+            #TOOD: what to do with the metafits_ppds file?
+            elif file.lower().endswith('00.fits'):
+                if '00' not in file_dict.keys():
+                    file_dict['00']=[file]
+                else:
+                    file_dict['00'].append(file)
+            elif file.lower().endswith('01.fits'):
+                if '01' not in file_dict.keys():
+                    file_dict['01']=[file]
+                else:
+                    file_dict['01'].append(file)
+            elif file.lower().endswith('.mwaf'):
+                if 'flags' not in file_dict.keys():
+                    file_dict['flags']=[file]
+                else:
+                    file_dict['flags'].append(file)        
+            else:
+                continue
+        
+        #checks:
+        if len(file_dict['00'])!=len(file_dict['01']):
+            raise ValueError('coarse band file missing')
+        if '01' not in file_dict.keys() and '00' not in file_dict.keys():
+            raise ValueError('no fits files submitted')
+        if 'flags' not in file_dict.keys():
+            warnings.warn('no flag files submitted')
+        
+        #first set parameters that are always true
+        self.Nspws=1
+        self.spw_array = np.array([0])
+        self.phase_type='drift'
+        self.vis_units='uncalib'
+        self.Npols=4 #TODO ask Bryna: should this be allowed to vary?
+        
+        #pull parameters from metafits file
+        with fits.open(metafits_file, memmap=True) as meta:
+            meta_hdr=meta[0].header
+            #get data from header
+            self.Nfreqs=meta_hdr['NCHANS']
+            self.Ntimes=meta_hdr['NSCANS']
+            self.channel_width=meta_hdr['FINECHAN']#this assumes no averaging by this code so will need to be updated
+            self.history = str(meta_hdr['HISTORY'])
+            if not uvutils._check_history_version(self.history, self.pyuvdata_version_str):
+                self.history += self.pyuvdata_version_str
+            self.integration_time=meta_hdr['INTTIME'] #TODO: talk to Bryna re: parameter definition)
```

Comment:
Integration time actually needs to be a vector of length Nblts (can all be the same value).